tools/loader/ImageClassifier.cpp

```diff
@@ -332,12 +332,37 @@ static void processAndPrintResults(Tensor *SMT, llvm::StringRef functionName) {
   }
 }
 
+/// If input file \p inputImageListFile is a text file then read all images
+/// listed in the text file
+static void parseInputImageList(const std::string &inputImageListFile) {
+  std::ifstream inFile;
+  inFile.open(inputImageListFile);
+  inputImageFilenames.erase(inputImageFilenames.begin());
+  while (!inFile.eof()) {
+    std::string img;
+    getline(inFile, img);
+    if (!img.empty()) {
+      inputImageFilenames.push_back(img);
+    }
+  }
+  inFile.close();
+}
+
 int main(int argc, char **argv) {
   Context ctx;
   // The loader verifies/initializes command line parameters, and initializes
   // the ExecutionEngine and Function.
   Loader loader(argc, argv);
 
+  std::string ext(".txt");
+  if (inputImageFilenames[0].rfind(ext) ==
```

Comment:
I'd just pass it as a separate arg (inputImagesListFile or something) and do not worry about specific extension etc.

Reply:
Since, inputImageFilenames has 'llvm::cl::OneOrMore'  attribute, I am obliged to reuse it. I am checking extension to differentiate between .png image files and list.txt file.

Just to clarify @rdzhabarov : Are you suggesting to create additional command line option 'inputImagesListFile'?

Reply:
Yeah, additional one. Just make inputImageFilenames to comply with 0 entries.